tennis-scheduler-web-app/src/components/TennisCourtAddChange.jsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from 'react'
+import { useEffect } from 'react';
+import { addTennisCourt, getTennisCourt, changeTennisCourt } from '../api/TennisCourtApi'
+import { useNavigate, useParams } from "react-router-dom";
+
+export const TennisCourtAddChange = () => {
+    const [address, setAddress] = useState({ country: "", city: "", street: "", number: "" })
+    const [surface, setSurface] = useState('')
+    const [image, setImage] = useState("")
+    const [tennisCourt, setTennisCourt] = useState({ name: "", surfaceType: "", description: "", image: "", address: address })
+    const [check, setCheck] = useState(false);
+    const [changedType, setChangedType] = useState('');
+    const [formErrors, setFormErrors] = useState({});
+    const [emptyForm, setEmptyForm] = useState(true);
+    const id = useParams().id
+    const navigate = useNavigate();
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setTennisCourt((tennisCourt) => ({ ...tennisCourt, [name]: value }));
+    };
+
+    const handleChangeAddress = e => {
+        const { name, value } = e.target;
+        setAddress((address) => ({ ...address, [name]: value }));
+    };
+
+    const setImages = e => {
+        const im = e.target.value.split("fakepath")[1].substring(1);
+        setImage(im)
+        setTennisCourt( tennisCourt => ({
+            ...tennisCourt,
+            image: im
+        }))
+        setCheck(true)
+    }
+
+    const onSubmit = e => {
+        e.preventDefault()
+        setTennisCourt( tennisCourt => ({
+            ...tennisCourt,
+            address: address
+        }))
+        setTennisCourt( tennisCourt => ({
+            ...tennisCourt,
+            surfaceType: surface
+        }))
+        setFormErrors(validation(tennisCourt))
+        if (Object.keys(formErrors).length === 0 && !emptyForm) {
+            if (id)
+                tennisCourtChange();
+            else
+                tennisCourtAdd();
+        }
+    }
+
+    const tennisCourtChange = () => {
+        changeTennisCourt(tennisCourt).then(() => {
+            navigate('/');
+            window.location.reload()
+        })
+    }
+
+    const tennisCourtAdd = () => {
+        addTennisCourt(tennisCourt).then(() => {
+            navigate('/');
+            window.location.reload()
+        })
```

Comment:
Here also.

Reply:
This add new tennis court.Calls endpoint and then go to page of all tennis courts.The same component is used for add new and for change existing tennis court.